src/oaklib/cli.py

```diff
@@ -202,13 +204,59 @@ def _shorthand_to_pred_curie(shorthand: str) -> PRED_CURIE:
         return shorthand
 
 
+# A list whose members are either strings (search terms, curies, or directives)
+# or nested lists.
+# TODO: Replace this with an explicit query model with boolean operations
+NESTED_LIST = Union[List[str], List["NESTED_LIST"]]
+
+
+def nest_list_of_terms(terms: List[str]) -> NESTED_LIST:
+    """
+    Gives a list of terms (typically passed on command line),
+    replace blocks between '[', ..., ']' with nested lists of the contents
+
+    :param terms:
+    :return:
+    """
+    nested, rest = _nest_list_of_terms(terms)
+    if rest:
+        raise ValueError(f"Unparsed: {rest}")
+    return nested
+
+
+def _nest_list_of_terms(terms: List[str]) -> Tuple[NESTED_LIST, List[str]]:
+    nested = []
+    while len(terms) > 0:
+        term = terms[0]
+        terms = terms[1:]
+        if term == "[":
+            nxt, rest = _nest_list_of_terms(terms)
+            terms = rest
+            nested.append(nxt)
+        elif term == "]":
+            return nested, terms
+        else:
+            nested.append(term)
+    return nested, []
```

Comment:
From here I get a sense that there is a possibility of nested lists. Would there be a disadvantage if the nested list was flattened first? 

`terms = list(itertools.chain(*terms))` or is it required to preserve the nesting in the list?

Reply:
consider also `list(itertools.chain.from_iterable(terms))`

Reply:
this is intentionally nested. There is an experimental feature where you can nest the boolean combinations of search directives. This will likely migrate to a full query datamodel rather than overloading lists